src/launch_server.c

```diff
@@ -0,0 +1,132 @@
+/*********************************** LICENSE **********************************\
+*                            Copyright 2017 Morphux                            *
+*                                                                              *
+*        Licensed under the Apache License, Version 2.0 (the "License");       *
+*        you may not use this file except in compliance with the License.      *
+*                  You may obtain a copy of the License at                     *
+*                                                                              *
+*                 http://www.apache.org/licenses/LICENSE-2.0                   *
+*                                                                              *
+*      Unless required by applicable law or agreed to in writing, software     *
+*       distributed under the License is distributed on an "AS IS" BASIS,      *
+*    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  *
+*        See the License for the specific language governing permissions and   *
+*                       limitations under the License.                         *
+\******************************************************************************/
+
+#include <builder.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/wait.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <arpa/inet.h>
+#include <netdb.h>
+#include <args.h>
+
+#define BACKLOG 10 /* How many pending connection we will keep in queue */
+
+// get sockaddr, IPv4 or IPv6:
+void *get_in_addr(struct sockaddr *sa)
+{
+    if (sa->sa_family == AF_INET)
+    {
+        return &(((struct sockaddr_in*)sa)->sin_addr);
+    }
+
+    return &(((struct sockaddr_in6*)sa)->sin6_addr);
+}
+
+void launch_server(void) {
+    int             sockfd = 0,  /* Read on this one */
+                    rem_fd = 0,  /* Fd that will be attributed to the client */
+                    rval;        /* To stock getaddrinfo() return */
+    s8_t            enable = 1;  /* Int to enable the option in setsockopt */
+    struct addrinfo hints,       /* Flags of getaddrinfo() */
+                    *servinfo,   /* Result of getaddrinfo() */
+                    *ptr;        /* Variable used to loop into the linked lst*/
+    char            port_str[6]; /* Buffer to hold the port number */
+
+    struct sockaddr_storage rem_addr;     /* Hold client infos */
+    socklen_t               sin_size = 0; /* Size of rem_addr */
+
+    char                    str[INET6_ADDRSTRLEN];
+
+    memset(&hints, 0, sizeof hints);
+    hints.ai_family = AF_UNSPEC;     /* Don't care IPv4 and IPv6 */
+    hints.ai_socktype = SOCK_STREAM; /* Use TCP stream socket */
+
+    /* Get port number as a string */
+    snprintf(port_str, 6, "%d", flags_get_port());
```

Comment:
Man snprintf sous debian : `The functions snprintf() and vsnprintf() write at most size  bytes  (including  the  terminating  null  byte ('\0')) to str.`
Sous OSX : `The snprintf() and vsnprintf() functions will write at most size-1 of the characters printed into the output string (the size'th character then gets the terminating `\0'); if the return value is greater than or equal to the size argument, the string was too short and some of the printed characters were discarded.  The output is always null-terminated, unless size is 0.`